src/seaborn_image/_core.py

```diff
@@ -106,24 +110,36 @@ def plot(self):
 
         if isinstance(self.cmap, str) and self.cmap in _CMAP_QUAL.keys():
             self.cmap = _CMAP_QUAL.get(self.cmap).mpl_colormap
-        
+
         if isinstance(self.cmap, str) and self.cmap in _CMAP_EXTRA.keys():
             self.cmap = _CMAP_EXTRA.get(self.cmap)
 
         if self.robust:
             min_robust = False
             max_robust = False
             if self.vmin is None:
-                min_robust = (
-                    True  # remember that vmin was None and now set to new value
-                )
+                # remember that vmin was None and now set to new value
+                min_robust = True
                 self.vmin = np.nanpercentile(self.data, self.perc[0])
             if self.vmax is None:
-                max_robust = (
-                    True  # remember that vmax was None and now set to new value
-                )
+                # remember that vmax was None and now set to new value
+                max_robust = True
                 self.vmax = np.nanpercentile(self.data, self.perc[1])
 
+        if self.diverging:
+            # Force vmin to have the same absolute value as vmax so that 0 is in the middle.
+
+            if self.vmaxabs is None:
+                if self.vmin is None or self.vmax is None:
+                    vmaxabs = np.abs(self.data).max()
+                else:
+                    vmaxabs = max(abs(self.vmin), abs(self.vmax))
+            else:
+                vmaxabs = self.vmaxabs
+
+            self.vmin = -vmaxabs
+            self.vmax = vmaxabs
```

Comment:
I would rewrite this so it does not require `vmaxabs` input from downstream functions (see my other comment).

```python
if self.vmax is None:
    self.vmax = np.abs(self.data).max()

if self.vmin is None:
    self.vmin = -self.vmax 
```

Another point is if a user passes either `vmin` or `vmax` or both, the expected behavior would be to respect the passed values for colorbar (even when `diverging=True`)